File: ios/App/App/WidgetBridgePlugin.swift
import Capacitor
import WidgetKit

// Pushes the web app's settings JSON into the App Group so widgets can
// resolve their "Same as app" options, then reloads all widget timelines.
@objc(WidgetBridgePlugin)
public class WidgetBridgePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "WidgetBridgePlugin"
    public let jsName = "WidgetBridge"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "syncSettings", returnType: CAPPluginReturnPromise)
    ]

    @objc func syncSettings(_ call: CAPPluginCall) {
        guard let settings = call.getString("settings") else {
            call.reject("settings is required")
            return
        }
        UserDefaults(suiteName: "group.com.verba.clock")?.set(settings, forKey: "settings")
        WidgetCenter.shared.reloadAllTimelines()
        call.resolve()
    }
}

File: android/app/src/main/java/com/verba/clock/WidgetBridgePlugin.java
package com.verba.clock;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.verba.clock.widget.VerbaWidgetProvider;
import com.verba.clock.widget.WidgetSettings;

/** Android twin of ios/App/App/WidgetBridgePlugin.swift: hands the app's settings to the widget. */
@CapacitorPlugin(name = "WidgetBridge")
public class WidgetBridgePlugin extends Plugin {
    @PluginMethod
    public void syncSettings(PluginCall call) {
        String settings = call.getString("settings");
        if (settings == null) {
            call.reject("settings is required");
            return;
        }
        WidgetSettings.store(getContext(), settings);
        VerbaWidgetProvider.refresh(getContext());
        call.resolve();
    }
}
